curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md

```diff
@@ -8,93 +8,53 @@ challengeType: 2
+With a headless browser, before the actual testing, we need to **visit** the page we are going to inspect. The `suiteSetup` 'hook' is executed only once at the suite startup. Other different hook types can be executed before each test, after each test, or at the end of a suite. See the Mocha docs for more information.
 
-# --hints--
+# --instructions--
 
-All tests should pass.
+Within `tests/2_functional-tests.js`, immediately after the `Browser` declaration, add your project URL to the `site` property of the variable:
 
 ```js
-(getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
-    (data) => {
-      assert.equal(data.state, 'passed');
-    },
-    (xhr) => {
-      throw new Error(xhr.responseText);
-    }
-  );
+Browser.site = 'https://sincere-cone.gomix.me'; // Your URL here
 ```
 
-You should assert that the headless browser request succeeded.
+If you are testing on a local environment replace the line above with
 
 ```js
-(getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
-    (data) => {
-      assert.equal(data.assertions[0].method, 'browser.success');
-    },
-    (xhr) => {
-      throw new Error(xhr.responseText);
-    }
-  );
+Browser.localhost('example.com', process.env.PORT || 3000);
 ```
 
-You should assert that the text inside the element 'span#name' is 'Amerigo'.
+Within `tests/2_functional-tests.js`, at the root level of the `'e2e Testing with Zombie.js'` suite, instantiate a new instance of the `Browser` object with the following code:
 
 ```js
-(getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
-    (data) => {
-      assert.equal(data.assertions[1].method, 'browser.text');
-      assert.match(data.assertions[1].args[0], /('|")span#name\1/);
-      assert.match(data.assertions[1].args[1], /('|")Amerigo\1/);
-    },
-    (xhr) => {
-      throw new Error(xhr.responseText);
-    }
-  );
+const browser = new Browser();
 ```
 
-You should assert that the text inside the element 'span#surname' is 'Vespucci'.
+Then, use the `suiteSetup` hook to direct the `browser` to the `/` route with the following code:
 
 ```js
-(getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
-    (data) => {
-      assert.equal(data.assertions[2].method, 'browser.text');
-      assert.match(data.assertions[2].args[0], /('|")span#surname\1/);
-      assert.match(data.assertions[2].args[1], /('|")Vespucci\1/);
-    },
-    (xhr) => {
-      throw new Error(xhr.responseText);
-    }
-  );
+suiteSetup(function(done) {
+  return browser.visit('/', done);
+});
 ```
 
-You should assert that the element 'span#dates' exist and its count is 1.
+# --hints--
+
+All tests should pass.
 
 ```js
 (getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
+  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=0').then(
```

Comment:
This just checks if the first functional test has passed. Do we not want it to check if all previous 4 are passing, instead?

Reply:
Sure - I've changed that 🙂